WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
```

Comment:
You know, all this complex diffing and updating can be replaced by just deleting all existing rounds and creating a new set from scratch. Much simpler and safer.

I believe this approach is what makes React so nice?